### DISCUSSION TITLE
LED Laser Module Hookup

### DISCUSSION THREAD

**Discussion by ic-racer** [srozum/film_camera_tester#34] [Q&A] [answered]
I have a 5V LED Laser for Sensor 3.  I don't have the KY008 'Module' with three pins. 

Apparently I need to connect either "Signal" or "Vcc" to the LED Laser with a 10K resistor. I found three ways to accomplish this. 

"A" and "B" seem the same and I'd think the LED Laser would be ON all the time, irrespective of "Signal."

"C" does not seem to make sense; shorting Vcc to Ground via the 10K resistor. 

"D" seems to make the most sense. 
![Schematic Composite LED Module](https://github.com/srozum/film_camera_tester/assets/128929440/4d7db882-919d-46b9-bbc4-87b3600a62e7)

**Comment by srozum** [comment, marked as answer]
Schematics A, B and D are identical because current limiting resistor is usually part of a laser assembly (you can see it soldered on tiny LED pcb).

The reason why 5V from power source is going through the 10K resistor is to supply additional current. Powering "Laser" LED directly from GPIO pin is not recommended since it consumes more current (30mA I guess) than Arduino GPIO is rated for (20mA, I guess). When powered, Laser glows a bit but don't ignites fully until turned on (supplied with additional current) by a signal from Arduino. That's ho KY-008 works.

Please, follow the schematic and repeat it as is. Don't overthink.

**Comment by ic-racer** [comment]
Great explanation, thank you!

**Comment by ic-racer** [comment]
Follow up here. I connected my LED Laser as indicated in "D" above and it functions as expected:

![DSC_0136](https://github.com/srozum/film_camera_tester/assets/128929440/5858ad80-3de7-40ef-bfd5-d111df4cacee)
